packages/clay-datepicker/src/ClayDatepicker.js

```diff
@@ -0,0 +1,384 @@
+import classnames from 'classnames';
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import React, {useState, useEffect, useRef} from 'react';
+
+import {useCurrentMonth, useCurrentTime, useWeeks} from './Hooks';
+
+import Button from './Button';
+import DateNavigation from './DateNavigation';
+import Day from './Day';
+import Dropdown from './Dropdown';
+import InputDate from './InputDate';
+import TimePicker from './TimePicker';
+import Week from './Week';
+import Weekday from './Weekday';
+import Weekdays from './Weekdays';
+
+/**
+ * ClayDatepicker component.
+ * @return {React.createElement}
+ */
+function ClayDatepicker({
+	ariaLabel,
+	dateFormat,
+	defaultValue,
+	expanded: initialExpanded,
+	firstDayOfWeek,
+	footerElement,
+	id,
+	initialMonth,
+	inputName,
+	months,
+	onValueChange,
+	placeholder,
+	spritemap,
+	time,
+	timeFormat,
+	useNative,
+	weekdaysShort,
+	years,
+}) {
+	/**
+	 * Indicates the current month rendered on the screen.
+	 * @type {!Date}
+	 */
+	const [currentMonth, setCurrentMonth] = useCurrentMonth(initialMonth);
+
+	/**
+	 * Indicates the time selected by the user.
+	 * @type {!Date}
+	 */
+	const [currentTime, setCurrentTime] = useCurrentTime(timeFormat);
+
+	/**
+	 * The day selected by the user.
+	 * @type {!Date}
+	 */
+	const [daySelected, setDaySelected] = useState(initialMonth);
+
+	/**
+	 * The input value.
+	 * @type {?(Date|undefined)}
+	 */
+	const [inputValue, setInputValue] = useState(undefined);
+
+	/**
+	 * An array of the weeks and days list for the current month
+	 * @type {!Array<Array>}
+	 */
+	const [weeks, setWeeks] = useWeeks(currentMonth, firstDayOfWeek);
+
+	/**
+	 * Flag to indicate if date is expanded.
+	 * @type {?bool}
+	 */
+	const [expanded, setExpanded] = useState(initialExpanded);
+
+	/**
+	 * Create a ref to store the datepicker DOM element
+	 */
+	const elementRef = useRef(null);
+
+	/**
+	 * Handles the click on element of the day
+	 * @param {!Date} date
+	 */
+	function handleDayClicked(date) {
+		setDaySelected(date);
+		setInputValue(date);
+	}
+
+	/**
+	 * Handles the input change.
+	 * @param {!Event} event
+	 */
+	function inputChange(event) {
+		const {value} = event.target;
+		const format = `${dateFormat} ${time ? timeFormat : ''}`;
```

Comment:
Might be nice to trim off the unwanted(?) trailing space in the event that `time` is falsey. eg.

```suggestion
		const format = `${dateFormat} ${time ? timeFormat : ''}`.trim();
```

Reply:
I think I'll do the same thing I did in **InputDate**, I think it's better than having to call `trim()` for this.

```javascript
const format = time ? `${dateFormat} ${time}` : dateFormat;
```